mitodl/micromasters
# Added coupon audit tables

#### What are the relevant tickets?
Fixes #2195 

#### What's this PR do?
Adds audit tables for the coupon models, and uses `save_and_log` in place of `save`.

#### How should this be manually tested?
- Create a 100% off coupon for some purchasable course run (or its course or program) in the Django admin and verify that `CouponAudit` was also updated. Make sure to set a coupon code for it.
- Attach a user to a coupon by POSTing to `/api/v0/coupons/<code>/users/`, then verify that `UserCouponAudit` was updated
- Purchase a course run with the coupon. Verify that `RedeemedCouponAudit` has a new entry
